### ISSUE TITLE
🐞 Bug report: Query in post widget (Single Post template). "Source = related" and "Terms = categories" is listing all the posts instead of posts from current post's category.

### ISSUE THREAD

**Issue by editor-acd** [elementor/elementor#18641]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

When i use "posts" widget in "Single Post" page builder, I am using Query to show a list of posts which are related to the current post's category. To do this I am using "Query" tab where I am have selected "Related" for source and "Categories" in term. However, I observe that all the posts are getting shown irrespective of the categories. It was not there before I updated the plugin. 

### Steps to reproduce

In Single-Post template, add "Posts"
1. Layout : skin -> classic, columns -> 1, Posts per page -> 30, Image Position -> None, Excerpt -> disabled, Read more -> disabled
2. Query : Source -> related, Term -> Categories, Fallback -> None, Order By -> Date, Order -> Desc.

Expectation is that it should show the list only for the category of current post. However, this is listing all the posts irrespective of the category limited by "posts per page".

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently using the steps above.

### System Info

					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: Percona Server (GPL), Release 23, Revision 500fcf5 v5.7.23-23
	PHP Version: 7.4.29
	PHP Memory Limit: 512M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 516M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.9.3
	Site URL: https://analogcircuitdesign.com
	Home URL: https://analogcircuitdesign.com
	WP Multisite: No
	Max Upload Size: 512 MB
	Memory limit: 40M
	Max Memory limit: 512M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: 0
	Admin Email: editor@analogcircuitdesign.com
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 2.5.0
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:99.0) Gecko/20100101 Firefox/99.0

== Active Plugins ==
	Elementor
		Version: 3.6.5
		Author: Elementor.com

	Elementor Pro
		Version: 3.7.0
		Author: Elementor.com

	Site Kit by Google
		Version: 1.74.0
		Author: Google

	WP Fastest Cache
		Version: 1.0.1
		Author: Emre Vona

	WP Fastest Cache Premium
		Version: 1.6.3
		Author: Emre Vona

	Yoast SEO
		Version: 18.8
		Author: Team Yoast

== Must-Use Plugins ==
	Elementor Safe Mode
		Version: 1.0.0
		Author: Elementor.com

== Elements Usage ==
	
	archive : 1
		archive-posts : 2
		heading : 1
	footer : 1
		form : 1
		heading : 6
		nav-menu : 3
		search-form : 1
		spacer : 2
		theme-site-logo : 1
	header : 1
		nav-menu : 2
		progress-tracker : 1
		search-form : 1
		theme-site-logo : 1
	popup : 1
		button : 15
		heading : 1
	section : 1
		image-box : 15
	widget : 0
		heading : 1
		image-box : 1
		social-icons : 1
		wp-widget-categories : 1
	wp-post : 120
		button : 4
		call-to-action : 9
		heading : 155
		html : 1
		image : 77
		image-carousel : 1
		menu-anchor : 1
		text-editor : 213
	wp-page : 3
		heading : 1
		posts : 1
		text-editor : 2
		wp-widget-categories : 1
	single-post : 1
		breadcrumbs : 1
		divider : 4
		heading : 2
		html : 1
		post-comments : 1
		post-navigation : 1
		posts : 1
		share-buttons : 1
		spacer : 3
		table-of-contents : 1
		theme-post-content : 1
		theme-post-excerpt : 1
		theme-post-featured-image : 1
		theme-post-title : 1
	error-404 : 1
		divider : 1
		heading : 1
		search-form : 1
		spacer : 1
		text-editor : 1

== Elementor Experiments ==
	Optimized DOM Output: Active by default
	Improved Asset Loading: Active by default
	Improved CSS Loading: Active by default
	Inline Font Icons: Inactive by default
	Accessibility Improvements: Active by default
	Import Export Template Kit: Active by default
	Additional Custom Breakpoints: Active by default
	Hide native WordPress widgets from search results: Active by default
	: Inactive by default
	Flexbox Container: Inactive by default
	Default to New Theme Builder: Active by default
	Hello Theme Header & Footer: Active by default
	Landing Pages: Active by default
	Color Sampler: Active by default
	Favorite Widgets: Active by default
	Admin Top Bar: Active by default
	Page Transitions: Active by default
	Notes: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default

== Log ==
	
PHP: showing 9 of 9PHP: 2022-03-23 14:42:58 [notice X 11015][/home2/analogci/public_html/wp-content/plugins/elementor/modules/dev-tools/deprecation.php::301] The $control_id argument is deprecated since version 3.5.0! [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-03-23 14:45:41 [notice X 367][/home2/analogci/public_html/wp-content/plugins/elementor/modules/dev-tools/deprecation.php::301] The $control_id argument is deprecated since version 3.5.0! [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> rest_error_handler()
#1: /home2/analogci/public_html/wp-content/plugins/elementor/modules/dev-tools/deprecation.php(301): class type trigger_error()
#2: /home2/analogci/public_html/wp-content/plugins/elementor/includes/managers/controls.php(523): Elementor\Modules\DevTools\Deprecation -> deprecated_argument()
#3: /home2/analogci/public_html/wp-content/plugins/elementor/includes/managers/controls.php(501): Elementor\Controls_Manager -> register()
#4: /home2/analogci/public_html/wp-content/plugins/elementor-pro/modules/query-control/module.php(826): Elementor\Controls_Manager -> register_control()
',
)]
PHP: 2022-03-23 14:57:47 [notice X 4][/home2/analogci/public_html/wp-content/plugins/elementor/core/experiments/experiments-reporter.php::89] Undefined index: title [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-03-23 18:18:56 [notice X 349][/home2/analogci/public_html/wp-content/plugins/elementor/modules/dev-tools/deprecation.php::301] The $control_id argument is deprecated since version 3.5.0! [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> rest_error_handler()
#1: /home2/analogci/public_html/wp-content/plugins/elementor/modules/dev-tools/deprecation.php(301): class type trigger_error()
#2: /home2/analogci/public_html/wp-content/plugins/elementor/includes/managers/controls.php(523): Elementor\Modules\DevTools\Deprecation -> deprecated_argument()
#3: /home2/analogci/public_html/wp-content/plugins/elementor/includes/managers/controls.php(501): Elementor\Controls_Manager -> register()
#4: /home2/analogci/public_html/wp-content/plugins/elementor-pro/modules/theme-builder/module.php(162): Elementor\Controls_Manager -> register_control()
',
)]
PHP: 2022-03-23 18:18:56 [notice X 350][/home2/analogci/public_html/wp-content/plugins/elementor/modules/dev-tools/deprecation.php::301] The $control_id argument is deprecated since version 3.5.0! [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> rest_error_handler()
#1: /home2/analogci/public_html/wp-content/plugins/elementor/modules/dev-tools/deprecation.php(301): class type trigger_error()
#2: /home2/analogci/public_html/wp-content/plugins/elementor/includes/managers/controls.php(523): Elementor\Modules\DevTools\Deprecation -> deprecated_argument()
#3: /home2/analogci/public_html/wp-content/plugins/elementor/includes/managers/controls.php(501): Elementor\Controls_Manager -> register()
#4: /home2/analogci/public_html/wp-content/plugins/elementor-pro/modules/forms/module.php(74): Elementor\Controls_Manager -> register_control()
',
)]
PHP: 2022-03-23 18:18:56 [notice X 350][/home2/analogci/public_html/wp-content/plugins/elementor/modules/dev-tools/deprecation.php::301] The $control_id argument is deprecated since version 3.5.0! [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> rest_error_handler()
#1: /home2/analogci/public_html/wp-content/plugins/elementor/modules/dev-tools/deprecation.php(301): class type trigger_error()
#2: /home2/analogci/public_html/wp-content/plugins/elementor/includes/managers/controls.php(523): Elementor\Modules\DevTools\Deprecation -> deprecated_argument()
#3: /home2/analogci/public_html/wp-content/plugins/elementor/includes/managers/controls.php(501): Elementor\Controls_Manager -> register()
#4: /home2/analogci/public_html/wp-content/plugins/elementor-pro/modules/forms/module.php(75): Elementor\Controls_Manager -> register_control()
',
)]
PHP: 2022-04-06 16:36:15 [notice X 2][/home2/analogci/public_html/wp-content/plugins/elementor/includes/managers/image.php::87] Trying to access array offset on value of type bool [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-04-21 12:30:08 [error X 4][/home2/analogci/public_html/wp-content/plugins/elementor/core/wp-api.php::26] Uncaught Error: Call to undefined function Elementor\Core\get_plugins() in /home2/analogci/public_html/wp-content/plugins/elementor/core/wp-api.php:26
Stack trace:
#0 /home2/analogci/public_html/wp-content/plugins/elementor/modules/compatibility-tag/base-module.php(96): Elementor\Core\Wp_Api->get_plugins()
#1 /home2/analogci/public_html/wp-content/plugins/elementor/modules/compatibility-tag/base-module.php(58): Elementor\Modules\CompatibilityTag\Base_Module->get_plugins_with_header()
#2 /home2/analogci/public_html/wp-content/plugins/elementor/modules/compatibility-tag/module.php(57): Elementor\Modules\CompatibilityTag\Base_Module->get_plugins_to_check()
#3 /home2/analogci/public_html/wp-content/plugins/elementor/modules/compatibility-tag/base-module.php(148): Elementor\Modules\CompatibilityTag\Module->get_plugins_to_check()
#4 /home2/analogci/public_html/wp-includes/class-wp-hook.php(307): Elementor\Modules\CompatibilityTag\Base_Module->Elementor\Modules\CompatibilityTag\{closure}('')
#5 /home2/analogci/publi [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-05-02 01:22:36 [notice X 13][/home2/analogci/public_html/wp-content/plugins/elementor/core/common/modules/ajax/module.php::171] Undefined index: data [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

Log: showing 20 of 302022-04-13 15:43:36 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.6.2',
  'to' => '3.6.3',
)]
2022-04-13 15:43:37 [info] elementor::elementor_updater Started 
2022-04-13 15:43:37 [info] Elementor/Upgrades - _on_each_version Start  
2022-04-13 15:43:38 [info] Elementor/Upgrades - _on_each_version Finished 
2022-04-13 15:43:38 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.6.2',
  'to' => '3.6.3',
)]
2022-04-13 20:00:47 [info] elementor::elementor_updater Started 
2022-04-13 20:00:47 [info] Elementor/Upgrades - _on_each_version Start  
2022-04-13 20:00:47 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.6.3',
  'to' => '3.6.4',
)]
2022-04-13 20:00:47 [info] Elementor/Upgrades - _on_each_version Finished 
2022-04-13 20:00:47 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.6.3',
  'to' => '3.6.4',
)]
2022-04-28 18:00:27 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.6.4',
  'to' => '3.6.5',
)]
2022-04-28 18:00:29 [info] elementor::elementor_updater Started 
2022-04-28 18:00:29 [info] Elementor/Upgrades - _on_each_version Start  
2022-04-28 18:00:32 [info] Elementor/Upgrades - _on_each_version Finished 
2022-04-28 18:00:32 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.6.4',
  'to' => '3.6.5',
)]
2022-05-11 16:49:33 [info] elementor-pro::elementor_pro_updater Started 
2022-05-11 16:49:33 [info] Elementor Pro/Upgrades - _on_each_version Start  
2022-05-11 16:49:33 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2022-05-11 16:49:33 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.5.2',
  'to' => '3.7.0',
)]
2022-05-11 16:49:33 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.5.2',
  'to' => '3.7.0',
)]

JS: showing 3 of 3JS: 2022-05-05 16:07:23 [error X 3][https://analogcircuitdesign.com/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.6.5:2:515477] T is null 
JS: 2022-05-13 17:17:29 [error X 34][https://analogcircuitdesign.com/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.6.5:2:235613] T.params.args.data is undefined 
JS: 2022-05-13 17:20:07 [error X 5][https://analogcircuitdesign.com/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.6.5:2:235643] Cannot read properties of undefined (reading \'id\') 

== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatibility not specified

== Elementor Pro - Compatibility Tag ==

**Comment by KNBAuthor** [comment]
I want to bump this because I'm having the exact same problem.

**Comment by tanyahayes** [comment]
Same here, very annoying, can't update to 3.7 because of this

**Comment by luciano77** [comment]
Same here, same problem.

**Comment by rosewordPress** [comment]
Same problem AND the main paige i.e. the homepage is repeating after the footer. The other pages menus are not working. I updated also to 3.7 - yesterday and this happened. Spoke to people who work at Elementor who didn't look at my error logs (not surprised lol) - then they sent it to Tier 2, who just looked at my backend, didn't look at the bugs being reported so now I have a website, that I can't do anything with. I even reinstalled the theme and still issues. I think back in the day of programming people did listen and we worked together to resolve the issue.
